modules/ducktests/src/main/java/org/apache/ignite/internal/ducktest/tests/rebalance/DataGenerationApplication.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.ducktest.tests.rebalance;
+
+import java.util.Random;
+import com.fasterxml.jackson.databind.JsonNode;
+import org.apache.ignite.IgniteCache;
+import org.apache.ignite.IgniteDataStreamer;
+import org.apache.ignite.configuration.CacheConfiguration;
+import org.apache.ignite.internal.ducktest.utils.IgniteAwareApplication;
+
+/**
+ *
+ */
+public class DataGenerationApplication extends IgniteAwareApplication {
+    /** Max streamer data size. */
+    private static final long MAX_STREAMER_DATA_SIZE = 100_000_000;
+
+    /** {@inheritDoc} */
+    @Override protected void run(JsonNode jsonNode) throws Exception {
+        int backups = jsonNode.get("backups").asInt();
+        int cacheCnt = jsonNode.get("cacheCount").asInt();
+        int entryCnt = jsonNode.get("entryCount").asInt();
+        int entrySize = jsonNode.get("entrySize").asInt();
+
+        log.info("Data generation started [backups=" + backups + ", cacheCount=" + cacheCnt
+            + ", entryCount=" + entryCnt + ", entrySize=" + entrySize + "]");
+
+        for (int i = 1; i <= cacheCnt; i++) {
+            IgniteCache<Integer, DataModel> cache = ignite.createCache(
```

Comment:
Let's use getOrCreateCache here. 
For a case target cache would be statically configured.

Reply:
done.